tests/integration-tests/tests/schedulers/test_slurm_accounting.py

```diff
@@ -0,0 +1,308 @@
+    dbusers = remote_command_executor.run_remote_command(
+        "sacctmgr list users DefaultAccount=pcdefault -Pn | cut -d '|' -f 1"
+    ).stdout
+    users = dbusers.splitlines()
+    for user in users:
+        logging.info("  User: %s", user)
+    assert_that(users).is_length(len(expected_users))
+    for user in users:
+        assert_that(user).is_in(*expected_users)
+
+
+@retry(stop_max_attempt_number=36, wait_fixed=10 * 1000)
+def _test_successful_startup_in_log(remote_command_executor):
+    log_file = "/var/log/slurmdbd.log"
+
+    log = remote_command_executor.run_remote_command("sudo cat {0}".format(log_file), hide=True).stdout
+    logging.info("slurmdbd.log: %s", log)
+    assert_that(
+        [line for line in log.splitlines() if STARTED_PATTERN.fullmatch(line) is not None], "Successful Startup"
+    ).is_not_empty()
+
+
+@retry(stop_max_attempt_number=36, wait_fixed=10 * 1000)
+def _test_slurmdbd_log_exists_in_log_group(cluster):
+    log_groups = cw_utils.get_cluster_log_groups_from_boto3(f"/aws/parallelcluster/{cluster.name}")
+    assert_that(log_groups).is_length(1)
+    log_group_name = log_groups[0].get("logGroupName")
+    log_streams = cw_utils.get_log_streams(log_group_name)
+    streams = [
+        stream.get("logStreamName")
+        for stream in log_streams
+        if re.fullmatch(r".*\.slurmdbd", stream.get("logStreamName")) is not None
+    ]
+    assert_that(streams).is_length(1)
+    stream_name = streams[0]
+    events = cw_utils.get_log_events(log_group_name, stream_name)
+    messages = (event.get("message") for event in events)
+    assert_that([message for message in messages if STARTED_PATTERN.fullmatch(message) is not None]).is_not_empty()
+
+
+def _test_getting_job_rows(remote_command_executor, expected_jobs):
+    for job in expected_jobs:
+        logging.info("Job: %s", job)
+        rows = remote_command_executor.run_remote_command(f"sacct -nP -j {job['job_id']}").stdout
+        logging.info("Rows: %s", rows)
+        for row in rows.splitlines():
+            columns = row.split("|")
+
+            job_id = columns[0].split(".")[0]
+            account = columns[3]
+            node_count = int(columns[4])
+            job_status = columns[5]
+
+            assert_that(job_id).is_equal_to(job["job_id"])
+            assert_that(account).is_equal_to(job["account"])
+            assert_that(node_count).is_equal_to(job["node_count"])
+            assert_that(job_status).is_equal_to("COMPLETED")
+
+
+def _test_for_no_connections(remote_command_executor, port):
+    assert_that(_get_connections(remote_command_executor, port)).is_empty()
+
+
+@pytest.mark.usefixtures("os", "scheduler")
+def test_slurm_accounting(
+    region,
+    instance,
+    pcluster_config_reader,
+    database_factory,
+    request,
+    test_datadir,
+    test_resources_dir,
+    clusters_factory,
+):
+    config_params = {
+        "compute_instance_type": instance,
+        "head_node_instance_type": instance,
+    }
```

Comment:
What is this for exactly? For possible re-using of these configurations in other scenarios?

Reply:
Originally I was thinking that we might want to support different head vs compute node instance types, probably not useful outside of perf testing though so I will remove this.

Reply:
Removed